django/core/cache/backends/memcached.py

```diff
@@ -105,37 +105,37 @@ def close(self, **kwargs):
         self._cache.disconnect_all()
 
     def incr(self, key, delta=1, version=None):
-        key = self.make_key(key, version=version)
-        self.validate_key(key)
+        made_key = self.make_key(key, version=version)
+        self.validate_key(made_key)
         # memcached doesn't support a negative delta
         if delta < 0:
-            return self._cache.decr(key, -delta)
+            return self.decr(key, -delta, version=version)
```

Comment:
Note that we pass `key` here, not `made_key`. Please let me know if you have better suggestions for the `made_key` variable 🙂

Reply:
Personally, I think `new_key` would be a more commonly used choice.

Reply:
Also, you might consider adding a 1-line comment before this line, for the same reason you made the comment here, e.g.

```python
# Pass the original key rather than the new key.
```

That way, readers of the code will know it's deliberate rather than an oversight.

Reply:
On second thought, it looks like you can just move the `make_key()` / `validate_key()` lines after the `delta < 0` block and not include a comment, since the new key isn't needed in that case. (Otherwise, you will be unnecessarily validating the original key twice when `delta < 0`.)

Reply:
True! I'll do that 🙂

Reply:
Force pushed into 0931dd244a48e8b75c24ac4b67fdee1c00d2014d